### DISCUSSION TITLE
Cache busting a script inside the public directory

### DISCUSSION THREAD

**Discussion by djlee0721** [vercel/next.js#20634] [Help] [answered]
I have a js file in the public folder (custom.js) for doing basic javascript stuff like animating the header, back to top button etc and I used _document.js to embed this script before the end of body tag.

Now in regards to **cache busting** - is there a way to **add hash** to the file name for cache busting?

Thank you very much in advance.

**Comment by jamesmosier** [comment, marked as answer]
There is no built-in way to do this in Next. The `public` folder is meant for static assets such as images or documents, rather than for serving JavaScript files. An alternative would be to import this file in your `_app.js` file and then it will be globally available:

```js
// assuming your move the `custom.js` file to a folder called `assets` (for example)
import '../assets/custom.js'

function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}
```

Then `custom.js` would be apart of the build and would then benefit from the normal caching/optimizations of bundles that Next provides out of the box.

**Reply by jamesmosier** [reply]
Thanks @danielLee0721, you too 🎉

**Reply by djlee0721** [reply]
Also one more thing..

Since I will be manipulating the DOM and the window object (doing things like animate nav bar on scroll.. back to top button etc).. I guess I will need to place the code inside a conditional like below?
```
if (typeof window !== "undefined") {
   // code goes here
  }

```
so that I don't get errors such as document is not defined or window is not defined?

**Reply by jamesmosier** [reply]
Correct, using `typeof window !== "undefined"` to conditionally execute browser logic is a great option.

**Reply by Mecanik** [reply]
Unfortunately, this method does not work anymore.
